### PR TITLE
Add build job for ARM64 Windows

### PR DESCRIPTION
**Pull request by windxu88**
## Purpose of this PR
Add build job for ARM64 Windows

CMake's `CMAKE_SYSTEM_PROCESSOR` variable returns `AMD64` even on ARM64 bokken VM. This is [a known bug](https://gitlab.kitware.com/cmake/cmake/-/issues/25466) in CMake.
So in this PR, I avoid using this variable. 
Instead, in `build.cmd` I query environment variable`PROCESSOR_ARCHITECTURE` and pass it to an user CMake variable `CMAKE_GENERATOR_PLATFORM`. 

In this PR, I also:
- Removed the prebuilt arm64 `abci.dll`.
- Remove 2023.2 from CI since it is not supported any more.

**Ticket/Jira #:**
[ABC-501](https://jira.unity3d.com/browse/ABC-501) Create Yamato job to build on Win ARM64

## Testing

**Functional Testing status:**

**Performance Testing status:**

## Overall Product Risks

**Complexity:**

**Halo Effect:**

## Documentation & UX Writing

| User facing text to review |  | Details |
| :--- | :--- | :--- |
| User interface |  |  |
| Public API docs |  |  |
| Changelog |  |  |

| Documentation halo effect |  | Jira link |
| :--- | :--- | :--- |
| User manual |  |  |
| Other docs |  |  |

## Additional information

**Note to reviewers:**

**Reminder:**

- [ ] Add entry in CHANGELOG.md

### CODE REVIEW

**Comment by windxu88** [comment]
Poke @thomas-tu @arejayelle for review

File: Source/abci/CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,8 @@ file(GLOB sources *.cpp *.h Foundation/*.h Foundation/*.cpp Importer/*.h Importe
 list(APPEND sources ${ISPC_SOURCES})
 list(APPEND sources ${ISPC_OBJECTS})
 
-if (CMAKE_SYSTEM_PROCESSOR STREQUAL "ARM64")
+message(STATUS "The value of CMAKE_GENERATOR_PLATFORM is: ${CMAKE_GENERATOR_PLATFORM}")
```
Comment (thomas-tu) [inline Source/abci/CMakeLists.txt:19]: Do you want to keep this log? (I'm okey with leaving it here, just want to double check)

File: cmake/FindISPC.cmake [hunk 2]
```diff
@@ -32,7 +32,7 @@ set(ISPC_VERSION 1.21.0)
             COMMAND ${7ZA} x -aoa -o* ${CMAKE_SOURCE_DIR}/External/ispc/${ISPC_DIR}.zip
         )
     endif()
-    cmake_print_variables(CMAKE_SYSTEM_PROCESSOR)
+    cmake_print_variables(CMAKE_GENERATOR_PLATFORM)
```
Comment (thomas-tu) [inline cmake/FindISPC.cmake:35]: Same question as above, do you want to keep this log?

**Review by thomas-tu — APPROVED** [review]

File: cmake/FindISPC.cmake (hunk 2 above)
Comment (windxu88) [reply cmake/FindISPC.cmake:35]: Let's keep it for now. I used it to confirm the build environment, whether AMD64 or ARM64. It is also no harm to have some log info in build process.

**Review by thomas-tu — APPROVED** [review]

**Pull request merged**